test/helpers/kube_dump.go

```diff
@@ -337,3 +485,26 @@ func fileAtPath(path string) *os.File {
 	}
 	return f
 }
+
+func writeControllerLog(ctx context.Context, outDir string, ns string, podName string, kubectlCli *kubectl.Cli) {
+	// Get the Gloo Gateway controller logs
+	controllerLogsFile := fileAtPath(filepath.Join(outDir, fmt.Sprintf("%s.controller.log", podName)))
+	controllerLogsCmd := kubectlCli.WithReceiver(controllerLogsFile).Command(ctx,
+		"-n", ns, "logs", podName, "-c", "gloo", "--tail=1000")
+	err := controllerLogsCmd.Run().Cause()
+	if err != nil {
+		fmt.Printf("error running controller logs for %s in %s command: %v\n", podName, ns, err)
+	}
+}
+
+func writeMetricsLog(ctx context.Context, outDir string, ns string, podName string, kubectlCli *kubectl.Cli) {
+	// Using an ephemeral debug pod fetch the metrics from the Gloo Gateway controller
+	metricsFile := fileAtPath(filepath.Join(outDir, fmt.Sprintf("%s.metrics.log", podName)))
```

Comment:
Does https://github.com/solo-io/gloo/blob/main/pkg/utils/kubeutils/kubectl/cli.go#L224 do what we need?

Reply:
I tried it and I'm getting an error (`error: Internal error occurred: unable to upgrade connection: container not found ("curl")`) related to it wanting a container named curl.